### PR TITLE
hdrop: init at 0.2.4

### PR DESCRIPTION
**Pull request by Schweber**
## Description of changes

[hdrop](https://github.com/Schweber/hdrop) is a helper script for [Hyprland](https://github.com/hyprwm/Hyprland) window manager that allows the user to run, show and hide specified programs via keybinds.

I am the author of `hdrop` and would also like to be its maintainer on nixpkgs.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/hd/hdrop/package.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  lib,
+  stdenvNoCC,
+  fetchFromGitHub,
+  makeWrapper,
+  scdoc,
+  coreutils,
+  util-linux,
+  jq,
+  libnotify,
+  hyprland ? null,
```
Comment (AndersonTorres) [inline pkgs/by-name/hd/hdrop/package.nix:11]: Never use null as default value.

```suggestion
  withHyprland ? true, hyprland
```

File: pkgs/by-name/hd/hdrop/package.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+}:
+stdenvNoCC.mkDerivation rec {
```
Comment (AndersonTorres) [inline pkgs/by-name/hd/hdrop/package.nix:13]:
```suggestion
}:

stdenvNoCC.mkDerivation rec {
```

File: pkgs/by-name/hd/hdrop/package.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  pname = "hdrop";
+  version = "0.2.3";
+
+  src = fetchFromGitHub {
+    owner = "Schweber";
+    repo = "hdrop";
+    rev = "9c29164";
```
Comment (AndersonTorres) [inline pkgs/by-name/hd/hdrop/package.nix:20]: Use full hash, or use something like 
```suggestion
    rev = v${version};
```

File: pkgs/by-name/hd/hdrop/package.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    sha256 = "sha256-aUL6mZ061rZBah8+/m5Guw23O7YeNK+FZY2obkVTk1I=";
```
Comment (AndersonTorres) [inline pkgs/by-name/hd/hdrop/package.nix:21]:
```suggestion
    hash = "sha256-aUL6mZ061rZBah8+/m5Guw23O7YeNK+FZY2obkVTk1I=";
```

File: pkgs/by-name/hd/hdrop/package.nix [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+    scdoc
+  ];
+
+  makeFlags = ["PREFIX=$(out)"];
```
Comment (AndersonTorres) [inline pkgs/by-name/hd/hdrop/package.nix:29]:
```suggestion
  makeFlags = [ "PREFIX=$(out)" ];
```

File: pkgs/by-name/hd/hdrop/package.nix [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+  postInstall = ''
+    wrapProgram $out/bin/hdrop --prefix PATH ':' \
+      "${lib.makeBinPath ([
+        coreutils
+        util-linux
+        jq
+        libnotify
+      ]
+      ++ lib.optional (hyprland != null) hyprland)}"
```
Comment (AndersonTorres) [inline pkgs/by-name/hd/hdrop/package.nix:39]:
```suggestion
      ++ lib.optional withHyprland hyprland)}"
```

File: pkgs/by-name/hd/hdrop/package.nix [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  '';
+
+  meta = with lib; {
+    description = "Emulate 'tdrop' in Hyprland (run, show and hide specific programs per keybind)";
+    homepage = "https://github.com/Schweber/hdrop";
+    changelog = "https://github.com/Schweber/hdrop/releases/tag/${version}";
+    license = licenses.agpl3;
+    platforms = platforms.unix;
+    maintainers = with maintainers; [Schweber];
```
Comment (AndersonTorres) [inline pkgs/by-name/hd/hdrop/package.nix:48]:
```suggestion
    maintainers = with maintainers; [ Schweber ];
```

File: pkgs/by-name/hd/hdrop/package.nix [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+      ]
+      ++ lib.optional (hyprland != null) hyprland)}"
+  '';
+
+  meta = with lib; {
+    description = "Emulate 'tdrop' in Hyprland (run, show and hide specific programs per keybind)";
+    homepage = "https://github.com/Schweber/hdrop";
+    changelog = "https://github.com/Schweber/hdrop/releases/tag/${version}";
+    license = licenses.agpl3;
+    platforms = platforms.unix;
```
Comment (AndersonTorres) [inline pkgs/by-name/hd/hdrop/package.nix:47]:
```suggestion
    platforms = platforms.linux;
```

Since some of your dependencies are Linux-only.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Squash the commits according to CONTRIBUTING guide.

**Comment by Schweber** [comment]
@AndersonTorres I implemented your corrections, thank you. I have a question if i may:

Is there a way to squash commits without having to download the whole repository and doing it locally? I'd really like to avoid that in the future. Is it possible to do it from the browser or just do it on merge?

**Comment by AndersonTorres** [comment]
Unfortunately I don't know. The GitHub web browser interface is too primitive.

Maybe the Desktop solution have something useful, but I never used it.

**Comment by Schweber** [comment]
Ok, thanks. So `nixpkgs` doesn't allow squash on merge?

**Comment by AndersonTorres** [comment]
No. It would transform the history in an undecipherable blob.

**Comment by Schweber** [comment]
I see, thanks for the answer.

**Comment by Schweber** [comment]
all checks passed

File: pkgs/by-name/hd/hdrop/package.nix [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "hdrop";
+  version = "0.2.3";
+
+  src = fetchFromGitHub {
+    owner = "Schweber";
+    repo = "hdrop";
+    rev = "${version}";
```
Comment (AndersonTorres) [inline pkgs/by-name/hd/hdrop/package.nix:21]:
```suggestion
    rev = version;
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]
acceptable

File: maintainers/maintainer-list.nix [hunk 10]
```diff
@@ -16113,6 +16113,11 @@
     githubId = 5104601;
     name = "schnusch";
   };
+  Schweber = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:16116]: Rebase the commits according to the commit guidelines

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Pull request closed without merging**